### PR TITLE
[backend] Trigger enrichment connectors with filters (#5371)

### PR DESCRIPTION
**Pull request by SouadHadjiat**
### Proposed changes

* Define connector_trigger_filters
* Filter enrichment connectors during entity creation
* Display and update connector trigger filters for internal enrichment connectors

### Related issues

* #5371 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

Different ways to test this feature:
- Create an external reference => should trigger the import external reference connector if it matches the filter
- OR try with rss feeds

To test this PR, you need to run import-document and import-external-reference connectors locally. For import-external-reference, you will need to install wkhtmltopdf https://wkhtmltopdf.org/downloads.html and in config.yml set
wkhtmltopdf_path: 'C:\\Program Files\\wkhtmltopdf\\bin\\wkhtmltopdf.exe'

When you test import-document, set your config with auto: true and validate_before_import: true, then make sure the workbench is created on the container of the external-ref.

Capture d'écran 2024-06-07 102859

### CODE REVIEW

**Comment by JeremyCloarec** [comment]
Just a detail, but maybe the filter box could be moved just a bit down? The 'Add filter' text goes a bit onto the 'Trigger filters' text when we click on the filter box
image

**Comment by JeremyCloarec** [comment]
What's the expected behavior when auto is set to true in import-external-reference? Does it still take into account the trigger filters?

**Comment by SouadHadjiat** [comment]
> What's the expected behavior when auto is set to true in import-external-reference? Does it still take into account the trigger filters?

When trigger filters are set, it overrides the "auto" configuration, it will always trigger automatically the connector based on the filters, whether it was set to manual or automatic.

File: opencti-platform/opencti-graphql/src/domain/enrichment.js
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/src/domain/enrichment.js]: shouldn't it be something like 
`conn.connector_trigger_filters ?? await isStixMatchConnectorFilter(context, element, conn.connector_trigger_filters) : conn.auto === true` ?
if we want to override connector auto config?

Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/domain/enrichment.js]: yes you're right, it's fixed now

**Comment by SouadHadjiat** [comment]
all comments have been fixed, waiting for review & approval

File: opencti-platform/opencti-graphql/src/domain/enrichment.js
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/src/domain/enrichment.js]: Is is intended that no filters means matching all, but filters empty means matching nothing?

File: opencti-platform/opencti-graphql/src/domain/connector.js
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/src/domain/connector.js]: Why is 'auto' a supported input key, since we currently only change the filters on the frontendside? Is it going to be added in the future?

File: opencti-platform/opencti-graphql/src/domain/connector.js
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/src/domain/connector.js:160]: I'm not sure how the cache currently works, but shouldn't resetCacheForEntity be called for connector entity type? During my tests, it looks like the filter isn't taken into account without having to restart the backend

File: opencti-platform/opencti-graphql/src/domain/enrichment.js
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/domain/enrichment.js]: oh right, I changed the logic, and I forgot about this part

File: opencti-platform/opencti-graphql/src/domain/connector.js
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/domain/connector.js]: yes I added this in prevision of being able to change the auto param directly in the front, but maybe I could remove it

File: opencti-platform/opencti-graphql/src/domain/connector.js
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/domain/connector.js:160]: the cache works by listening to update events, which are sent here `notify(BUS_TOPICS[ENTITY_TYPE_CONNECTOR].EDIT_TOPIC, element, user);`

File: opencti-platform/opencti-graphql/src/domain/connector.js
Comment (JeremyCloarec) [reply opencti-platform/opencti-graphql/src/domain/connector.js]: I think it's OK to keep, but I was mostly wondering how it would work with the auto coming from the connector config file? Won't the auto configured with this method be overriden everytime the connector is restarted and registers to the API with its config from config file?

Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/domain/connector.js]: done

File: opencti-platform/opencti-graphql/src/domain/enrichment.js
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/domain/enrichment.js]: done

**Review by JeremyCloarec — APPROVED** [review]

**Pull request merged**
